### DISCUSSION TITLE
Integrate with Apple HomeKit through Home Assistant

### DISCUSSION THREAD

**Discussion by Patrick1610** [tsightler/ring-mqtt#715] [General Discussion]
Hi! I don't know if this is the right place to ask, but I am looking for some help regarding the following setup:  

I have a Ring Pro, Home Assistant and Apple Home. From Home Assistant I can expose camera's to Homekit to automatically stream doorbells on AppleTV on a Ding event. This is a standard feature of HomeKit and Home Assistant allows to expose devices to HomeKit. I have been able to add Ring-MQTT successfully to Home Assistant, and I have created the Generic Camera using the links. This works perfectly! Also, when adding the Generic Camera to HomeKit I can live-view the stream there. 

My issue is as follows: On the Ring-MQTT (ring_pro_snapshot) device, I don't get a live stream, but on the Generic Camera I do. Only, in HomeKit the Generic camera is not shown as a doorbell with the option to stream on events, while the ring_pro_snapshot device is. 

How can I combine the two correctly so that I can stream to AppleTV on a ding from my Ring device? 
I understand that this is quite a specific situation. If any more information is needed please let me know! 

With kind regards, 
Patrick

**Comment by tsightler** [comment]
I'm going to switch this to a discussion, maybe somebody in the community will have some input, however, as this project is for ring-mqtt, and your question is really about Homekit and Home Assistant, I'm not sure.

I would personally suggest using a tool like Homebridge or Scrypted for your Homekit needs vs attempting to gateway from ring-mqtt -> Home Assistant -> Homekit.

**Reply by Patrick1610** [reply]
I understand @tsightler! It seems that scrypted could be the way to go! Thanks for leaving this open as a discussion and pointing me in the right direction 🙏🏼

**Reply by tsightler** [reply]
Just to answer the original question, even if I still lean toward my suggestion to use something like Scrypted instead, if the ring-mqtt + Home Assistant Homekit bridge integration is otherwise good enough for you, then you should look at the "linked_doorbell_sensor" configuration option  By default Home Assistant will not know what generic camera goes with what doorbell sensor, but with this option it is possible to manually link the camera with the correct doorbell sensor entity and it will be exposed as a doorbell and you will get the behavior you want.  Good luck!

**Comment by Greminn** [comment]
@Patrick1610 - you should look into adding [scrypted](https://github.com/koush/scrypted) to your network (it can run as an HAOS Add-On), add the cameras into that. You will then get exactly what you want including HomeKit Secure Video support. I do this and use [ring-mqtt](https://github.com/tsightler/ring-mqtt) to bring things into HA. In this case I dont expose the cameras from HA to HomeKit as [scrypted](https://github.com/koush/scrypted) is doing it for me.

**Reply by Greminn** [reply]
Lol - did not even read @tsightler's full reply here before replying. Oh well, ill leave it as it is slightly more verbose :)

**Reply by Patrick1610** [reply]
Hi @Greminn, thanks for your reply! That sound like it could be exactly what I need! Could you tell me more about how you configured this? You created the camera in Home Assistant using Ring-mqtt? And how could this be easily added to scrypted? Or is it very complicated to set this up? I will definitely be playing with this coming weekend! Kind regards!

**Reply by Greminn** [reply]
No - scrypted is completely separate to Ring-mqtt. You configure the camera/doorbell in scrypted and then it exposes it to HomeKit. Read up here. they have an offical HA add-on to run it. https://docs.scrypted.app

**Reply by Greminn** [reply]
Note you dont need the NVR part (paid app) if you only want to use your home app to see video.

**Reply by Ramias1** [reply]
Piling on.  Run Scrypted in parallel for Ring to HomeKit Camera/Doorbell.  Scrypted also gives homekit secure video if you have a monthly Apple plan.  I don't use HKSV for my Ring cameras but I could; I have other cameras I do HKSV with from Scrypted.  You can then expose the Scrypted Cameras to Home Assistant for its purposes (but I wouldn't then to HA to Apple; I'd do Scripted to Apple and Scripted to HA)

**Reply by eirabie** [reply]
@Greminn if I understand this correctly, you are using ring-mqtt to bring cameras to HA and Scrypted to bring them to HomeKit? Does that mean you are logged into your Ring account from ring-mqtt and Scrypted separately?

**Reply by Greminn** [reply]
@eirabie im using ring-mqt to bring everything except video into HA, im pretty sure im logged into ring-mqt. Then video comes into HA via Scrypted, and im logged in there. So to answer your question: Yes!
